spec/helpers/log4js-logger-provider.ts

```diff
@@ -2,15 +2,10 @@ import { getLogger } from 'log4js';
 import { Logger, LoggerProvider } from '../../src/config/logging';
 
 export class Log4jsLoggerProvider implements LoggerProvider {
-    constructor(
-        public readonly level: string,
-        public readonly levelByCategory: { [category: string]: string } = {},
-    ) {}
-
     getLogger(category: string): Logger {
-        const logger = getLogger(category);
-        logger.level =
-            category in this.levelByCategory ? this.levelByCategory[category] : this.level;
-        return logger;
+        // note: we used to change the level here, but that does not work because in log4js,
+        // everything is global state (you're actually changing a category's level globally if you
+        // set .level on a logger)
+        return getLogger(category);
```

Comment:
You can also remove the constructor since the fields were only used in the function that you modified.